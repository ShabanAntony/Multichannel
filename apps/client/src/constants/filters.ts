export enum ItemFilters {
  DISCOUNT_50_60,
  DISCOUNT_60_PLUS,
}
