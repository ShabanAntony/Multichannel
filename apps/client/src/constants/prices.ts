export const laryToUSD = 2.7
export const VSProfit = 0.5
export const nikeProfit = 0.3
export const reebokProfit = 0.3
export const customsTax = 0.18 + 0.02
