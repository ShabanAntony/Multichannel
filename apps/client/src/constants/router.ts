export enum Pages {
  ITEMS = '/multistream/items/:brandType',
}

export enum BrandTypes {
  Classic = 'vs',
  PINK = 'pink',
  NIKE = 'nike',
  REEBOK = 'reebok',
}
