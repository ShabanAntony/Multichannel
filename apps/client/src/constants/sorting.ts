export enum ItemSorting {
  BY_MAX_DISCOUNT,
  BY_MIN_PRICE,
}
