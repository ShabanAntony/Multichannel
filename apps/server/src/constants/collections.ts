export const collectionsToCheckout = [
  // all sale
  'a202a327-de9d-40ac-8dc3-743ce96b870c',
  // panties
  'f5a391f6-2ca9-4ac4-ba2c-64caa6338185',
  // lingerie
  '7508da5f-a50e-4541-ab89-6741846b3606',
  // bras
  'b60bdcec-c132-48c6-8760-49fbd1446674',
  // sport
  '297255c3-5614-4009-ad7d-557378e8818f',
  // Beauty
  '7eaeaf81-f067-430e-80eb-78a5e871d479',
  // Bags & Accessories
  'cd0dd393-bffe-45e2-a899-6e2d243ceff8',
  // brand we love
  '41d80710-9f25-4405-a309-c2ae3388dc09',
]
