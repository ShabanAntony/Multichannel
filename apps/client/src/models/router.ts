import { Location } from 'react-router-dom'

import { createEvent, createStore } from 'effector'

import { BrandTypes } from '../constants/router'

export type Params = Partial<{
  brandType: BrandTypes
}>

export const setParams = createEvent<Params>('set url params')
export const $params = createStore<Params>({}).on(
  setParams,
  (_, params) => params,
)

export const setLocation = createEvent<Location>('set url params')
export const $location = createStore<Location>({
  hash: '',
  key: '',
  pathname: '',
  search: '',
  state: undefined,
}).on(setLocation, (_, params) => params)

// $location.watch((location) => {
//   console.log(`>> location: ${JSON.stringify(location, null, 4)}`)
// })
