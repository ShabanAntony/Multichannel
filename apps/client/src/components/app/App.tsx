import { FC, useEffect } from 'react'
import {
  Navigate,
  Route,
  Routes,
  generatePath,
  useLocation,
  useParams,
} from 'react-router-dom'

import { BrandTypes, Pages } from '../../constants/router'
import { Params, setLocation, setParams } from '../../models/router'
import { NikePage } from '../../pages/NikePage/NikePage'
import { PinkPage } from '../../pages/PinkPage/PinkPage'
import { ReebokPage } from '../../pages/ReebokPage/ReebokPage'
import { VSPage } from '../../pages/VSPage/VSPage'

export interface Props {}
export const App: FC<Props> = () => {
  const params = useParams<Params>()

  useEffect(() => {
    setParams(params)
  }, [params])

  const location = useLocation()

  useEffect(() => {
    setLocation(location)
  }, [location])

  return (
    <Routes>
      <Route
        path={generatePath(Pages.ITEMS, { brandType: BrandTypes.Classic })}
        element={<VSPage />}
      />
      <Route
        path={generatePath(Pages.ITEMS, { brandType: BrandTypes.PINK })}
        element={<PinkPage />}
      />
      <Route
        path={generatePath(Pages.ITEMS, { brandType: BrandTypes.NIKE })}
        element={<NikePage />}
      />
      <Route
        path={generatePath(Pages.ITEMS, { brandType: BrandTypes.REEBOK })}
        element={<ReebokPage />}
      />
      <Route
        path="*"
        element={
          <Navigate
            to={generatePath(Pages.ITEMS, { brandType: BrandTypes.Classic })}
          />
        }
      />
    </Routes>
  )
}
