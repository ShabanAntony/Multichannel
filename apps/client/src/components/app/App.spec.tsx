import { MemoryRouter, generatePath } from 'react-router-dom'
import renderer from 'react-test-renderer'

import { BrandTypes, Pages } from '../../constants/router'
import { App } from './App'

jest.mock('../../pages/VSPage/VSPage', () => ({
  VSPage: 'mocked-VSPage',
}))

jest.mock('../../pages/PinkPage/PinkPage', () => ({
  PinkPage: 'mocked-PinkPage',
}))

jest.mock('../../pages/NikePage/NikePage', () => ({
  NikePage: 'mocked-NikePage',
}))

describe('App', () => {
  it('should render default router', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render VS classic route', () => {
    const tree = renderer
      .create(
        <MemoryRouter
          initialEntries={[
            generatePath(Pages.ITEMS, { brandType: BrandTypes.Classic }),
          ]}
        >
          <App />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render PINK route', () => {
    const tree = renderer
      .create(
        <MemoryRouter
          initialEntries={[
            generatePath(Pages.ITEMS, { brandType: BrandTypes.PINK }),
          ]}
        >
          <App />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render NIKE route', () => {
    const tree = renderer
      .create(
        <MemoryRouter
          initialEntries={[
            generatePath(Pages.ITEMS, { brandType: BrandTypes.NIKE }),
          ]}
        >
          <App />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
