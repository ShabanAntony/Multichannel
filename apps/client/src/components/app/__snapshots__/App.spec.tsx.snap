// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render NIKE route 1`] = `<mocked-NikePage />`;

exports[`App should render PINK route 1`] = `<mocked-PinkPage />`;

exports[`App should render VS classic route 1`] = `<mocked-VSPage />`;

exports[`App should render default router 1`] = `null`;
